moshi/src/main/java/com/squareup/moshi/JsonWriter.java

```diff
@@ -335,6 +336,8 @@ public final void setSerializeNulls(boolean serializeNulls) {
    */
   public abstract JsonWriter value(@Nullable Number value) throws IOException;
 
+  public abstract JsonWriter value(BufferedSource source) throws IOException;
```

Comment:
This is a clever API. Nice.

Reply:
I'll write up a Javadoc since the API looks good to you!

Reply:
I considered valueRaw(String) as a convenience API for Buffer.writeUtf8(), but I'm hoping this is used much more for appending from non-memory sources.

Reply:
Yeah, offering anything that takes a String is probably a disservice; users will call that thinking it’s the same as `value(String)`